R/utils-fs.R

```diff
@@ -110,3 +110,16 @@ pkgdown_config_relpath <- function(pkg) {
 
   fs::path_rel(config_path, pkg$src_path)
 }
+
+# Will error if parent path doesn't exist.
+# https://github.com/r-lib/pkgdown/issues/2329
+check_dst_path_exists <- function(path, call = caller_env(), check = is_interactive()) {
+  if (check && !dir_exists(path)) {
+    cli::cli_abort(c(
+      "Can't create a site",
+      i = "Do you need to run {.run pkgdown::init_site()}?"
+    ),
+    call = call)
```

Comment:
```suggestion
    init_site()
```

I see. The problem is that `init_site()` is not very fast. Ideally, we would just call it when necessary.

And I think that the interactive condition has to remain there, otherwise this would break tests (or make them significantly slower)